### PR TITLE
Update clang-dist-toolchain [ci full]

### PR DESCRIPTION
**Pull request by skhamis**
### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due diligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by DonalMe** [comment]
@mergify backport release-v130

**Review by bendk — APPROVED** [review]
Changes look good to me, assuming the CI passes.

@skhamis Should we make another one of these for `cctools.tar.zst`, a couple of lines above?  I think it should be in a separate PR since there's less urgency to uplift it.

**Comment by skhamis** [comment]
> Changes look good to me, assuming the CI passes.
> 
> @skhamis Should we make another one of these for `cctools.tar.zst`, a couple of lines above? I think it should be in a separate PR since there's less urgency to uplift it.

Looks like it was still failing with the updated clang-dist, and cc-tools also 404's, so i'm gonna update here as that was also probably a problem.

File: taskcluster/scripts/toolchain/cross-compile-setup.sh [hunk 1]
```diff
@@ -27,11 +27,11 @@ export TARGET_CFLAGS="-DNDEBUG"
...
 curl -sfSL --retry 5 --retry-delay 10 \
-    https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/gecko.cache.level-3.toolchains.v3.linux64-cctools-port.hash.bec815dd1e64cd75ba803c1f002c646e151c784d804791bffbc23b2ae41f545f/artifacts/public/build/cctools.tar.zst > cctools.tar.zst
+    https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/gecko.cache.level-3.toolchains.v3.linux64-cctools-port.pushdate.2024.07.23.20240723071212/artifacts/public%2Fbuild%2Fcctools.tar.zst > cctools.tar.zst
 tar -I zstd -xf cctools.tar.zst
 rm cctools.tar.zst
 curl -sfSL --retry 5 --retry-delay 10 \
-    https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/gecko.cache.level-3.toolchains.v3.clang-dist-toolchain.hash.9e5dd392eefd9eb1ef3ff7aa40d2823f60b280cb54d716e0f85f0b50573e0d69/artifacts/public/build/clang-dist-toolchain.tar.xz > clang-dist-toolchain.tar.xz
+    https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/gecko.cache.level-3.toolchains.v3.clang-dist-toolchain.pushdate.2024.07.30.20240730145721/artifacts/public%2Fbuild%2Fclang-dist-toolchain.tar.xz > clang-dist-toolchain.tar.xz
 tar -xf clang-dist-toolchain.tar.xz
 mv builds/worker/toolchains/clang clang
```
Comment (bendk) [inline taskcluster/scripts/toolchain/cross-compile-setup.sh:36]:
```suggestion
mv builds/worker/toolchains/clang/bin/clang clang
```

Comment (bendk) [reply taskcluster/scripts/toolchain/cross-compile-setup.sh:36]: I think this might work.  It looks like the location of `clang` changed.

**Review by linabutler — APPROVED** [review]
Ship it!

**Pull request merged**